internal/handlers/common/projection.go

```diff
@@ -85,34 +108,114 @@ func ProjectDocuments(docs []*types.Document, projection *types.Document) error
 		return err
 	}
 
-	projectionMap := projection.Map()
 	for i := 0; i < len(docs); i++ {
-		for k := range docs[i].Map() {
-			v, ok := projectionMap[k]
-			if !ok {
-				if k == "_id" { // if _id is not in projection map, do not do anything with it
-					continue
-				}
-				if inclusion {
-					docs[i].Remove(k)
-				}
+		err = projectDocument(inclusion, docs[i], projection)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func projectDocument(inclusion bool, doc *types.Document, projection *types.Document) error {
+	projectionMap := projection.Map()
+
+docMapLoopK1:
```

Comment:
Let's refactor this function to reduce cyclomatic complexity and to avoid loop labels. I think some functions can be extracted

Reply:
done, thank you for note.